Sources/HTTP/HTTPServer.swift

```diff
@@ -28,31 +31,72 @@ public class HTTPServer {
     /// To process incoming requests
     public let handler: HTTPRequestHandler
 
-    private let server = PoCSocketSimpleServer()
+    /// Stop the server
+    public func stop() {
+        server?.close()
+    }
+
+    /// The port number the server is listening on
+    public var port: Int {
+        return options.port
+    }
+
+    /// Reference to the server for closing
+    private var server: TCPServer?
+
+    /// The number of current connections
+    public var connectionCount: Int
 
     /// Create an instance of the server. This needs to be followed with a call to `start(port:handler:)`
     public init(with newOptions: Options, requestHandler: @escaping HTTPRequestHandler) {
         options = newOptions
         handler = requestHandler
+        connectionCount = 0
     }
 
     /// Start the HTTP server on the given `port` number, using a `HTTPRequestHandler` to process incoming requests.
     public func start() throws {
-        try server.start(port: options.port, handler: handler)
+        let socket = try TCPSocket(isNonBlocking: true, shouldReuseAddress: true)
+        var eventLoops: [DispatchQueue] = []
+        for i in 1...8 {
```

Comment:
Why 8? :-)

Reply:
there are quite a few places i used magic numbers while porting this over. (like the accept queue, and read size as well) those can be easily fixed if people like this